src/VisualStudio/Core/Def/ValueTracking/ValueTrackingTreeViewModel.cs

```diff
@@ -83,6 +89,17 @@ public ValueTrackingTreeViewModel(IClassificationFormatMap classificationFormatM
             };
 
             PropertyChanged += Self_PropertyChanged;
+            _solutionEventMonitor.SolutionClosed += OnSolutionClosed;
+        }
+
+        public void Dispose()
+        {
+            _solutionEventMonitor.SolutionClosed -= OnSolutionClosed;
+        }
+
+        private void OnSolutionClosed(object sender, EventArgs e)
+        {
+            Roots.Clear();
```

Comment:
It doesn't look like we're doing much here, but I know we've had problems with doing work on solution close.  Should we potentially instead do this on solution open instead?

Reply:
Ideally not, just because we are leaving things around in memory that are not cheap. Can add a note about this being a performance critical path. It is thread dependent (ObservableCollection) but we could also queue work off thread and jump back?

Reply:
does thsi cancel any outstanding work?  or is it still happening, albeit invisibly?  i'm fine with eiter, but it would be nice to doc what hte semantics ar ehere.

Reply:
It does not cancel any outstanding work. That will be handled by cancellation from other sources. This just clears the memory entirely.

Reply:
consider for the future adding a CTS that you can trigger here to quickly stop that work.  Doesn'ts eem blocking for thsi PR though :)